core/trino-spi/src/main/java/io/trino/spi/exchange/ExchangeManagerContext.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.exchange;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+
+public interface ExchangeManagerContext
+{
+    default OpenTelemetry getOpenTelemetry()
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    default Tracer getTracer()
+    {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
remove default implementations. the implementations are supposed to be complete

Reply:
This is the same pattern `ConnectorContext`. The reason for having this as an interface (as well as migrating `ExchangeContext` to interface) is that otherwise we would need to make trino-spi depend on parts of `opentelemetry` implementation `opentelemetry-context`. Not only `opentelemetry-api`.